src/devices/cpu/arcompact/arcompact_execute.cpp

```diff
@@ -2549,7 +2568,10 @@ ARCOMPACT_RETTYPE arcompact_device::arcompact_handle04_05_p00(OPS_32)
 
 	if (F)
 	{
-		arcompact_fatal("arcompact_handle04_05 (OR) (F set)\n"); // not yet supported
+	        if (result & 0x80000000) { STATUS32_SET_N; }
+		else { STATUS32_CLEAR_N; }
+		if (result == 0x00000000) { STATUS32_SET_Z; }
+		else { STATUS32_CLEAR_Z; }
```

Comment:
Same with this – you aren’t matching the style of anything else in the file, and it doesn’t make the code clearer.

Also, given the number of times you’ve reproduced this code, wouldn’t it make more sense to factor it out into an inline member function?